src/function/cast_string_non_nested_functions.cpp

```diff
@@ -172,6 +172,9 @@ LogicalType inferMinimalTypeFromString(std::string_view str) {
     if (cpy.size() == 0) {
         return LogicalType::ANY();
     }
+    if (cpy == "NULL") {
```

Comment:
I prefer not adding a new datatype, instead we treat it as `LogicalType::ANY()`

Reply:
done this